### ISSUE TITLE
When opening a folder for the first time use the Default profile

### ISSUE THREAD

**Issue by isidorn** [microsoft/vscode#169260]
Currently when I open a folder for the first time we use the currently active profile.
I believe this can be confusing as the user has not explicitly picked the profile they want to use for the new folder.

Based on @jrieken suggestion let's use the default profile for folders that are opened for the very first time.

I believe this with giving users an option to chose which is the default https://github.com/microsoft/vscode/issues/168867 would give users enough options and clarity when interacting with profiles.

fyi @stevencl

**Comment by bpasero** [comment]
I had reported https://github.com/microsoft/vscode/issues/168551 and feel this flow is no longer supported:
* use a non-default profile to install docker and container remote extension
* from the docker view open a container as new window
* 🐛 it will fail because the new window does not have the extension installed with default profile

Maybe we should make a decision based on how the user opened a new window? E.g. if an extension triggers this, always keep the profile?

**Comment by isidorn** [comment]
@bpasero good point. And thinking about this more this might have been a bad suggestion on my end.

I also think of the following - a new user just installed VS Code, and installed a Python profile. 
They are opening a new folder - and they get switched back to the default profile 🐛 

I think we should either:
1.  go back to the old behaviour and make sure to make it clearer what profile is applied to what folder to decrease @jrieken confusion
2. keep the current behaviour but make it easier for users to choose what is the "default profile". And cover cases like @bpasero brought up

@sandy081 what do you think?
